### PR TITLE
Update FilterPoliCheckResults.ps1

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Update FilterPoliCheckResults.ps1

### CODE REVIEW

File: eng/common/scripts/FilterPoliCheckResults.ps1 [hunk 1]
```diff
@@ -6,35 +6,42 @@ This script will read data speciefied in one or more PoliCheckAllowList.yml file
...
 )
 
 . "${PSScriptRoot}\logging.ps1"
 
-$RepoRoot = Resolve-Path -Path "${PSScriptRoot}\..\..\..\"
-$PathToAllowListFiles = Join-Path $RepoRoot $ServiceDirtectory
-$PolicCheckAllowListFiles = Get-ChildItem -Path $PathToAllowListFiles -Recurse -File -Include "PoliCheckAllowList.yml"
+# Install Powershell Yaml
+$ProgressPreference = "SilentlyContinue"
+$toolsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-tools/nuget/v2"
```
Comment (benbp) [inline eng/common/scripts/FilterPoliCheckResults.ps1:30]: If this script is going to be common and presumably used for multiple repos, we probably want to pass the feed in as a parameter from the calling pipelines in the language repos.

File: eng/common/scripts/FilterPoliCheckResults.ps1 [hunk 2]
```diff
@@ -6,35 +6,42 @@ This script will read data speciefied in one or more PoliCheckAllowList.yml file
...
-    $allowListData["PC1002"] += $allowListDataInFile["PC1002"]
-    $allowListData["PC1003"] += $allowListDataInFile["PC1003"]
-    $allowListData["PC1004"] += $allowListDataInFile["PC1004"]
-    $allowListData["PC1005"] += $allowListDataInFile["PC1005"]
-    $allowListData["PC1006"] += $allowListDataInFile["PC1006"]
+    $allowListData = ConvertFrom-Yaml (Get-Content $allowListFilePath -Raw)
+}
+else
+{
+    LogError "Allow list path ${allowListFilePath} does not exisit."
```
Comment (benbp) [inline eng/common/scripts/FilterPoliCheckResults.ps1:43]:
```suggestion
    LogError "Allow list path ${allowListFilePath} does not exist."
```

File: eng/common/scripts/FilterPoliCheckResults.ps1 [hunk 3]
```diff
@@ -6,35 +6,42 @@ This script will read data speciefied in one or more PoliCheckAllowList.yml file
 It then reamoves all allwed entries from the PoliCheckResult
```
Comment (benbp) [inline eng/common/scripts/FilterPoliCheckResults.ps1:6]: While you're in here fixing spelling:

```suggestion
It then removes all allowed entries from the PoliCheckResult 
```

File: eng/common/scripts/FilterPoliCheckResults.ps1 [hunk 4]
```diff
@@ -6,35 +6,42 @@ This script will read data speciefied in one or more PoliCheckAllowList.yml file
...
-$RepoRoot = Resolve-Path -Path "${PSScriptRoot}\..\..\..\"
-$PathToAllowListFiles = Join-Path $RepoRoot $ServiceDirtectory
-$PolicCheckAllowListFiles = Get-ChildItem -Path $PathToAllowListFiles -Recurse -File -Include "PoliCheckAllowList.yml"
+# Install Powershell Yaml
+$ProgressPreference = "SilentlyContinue"
+$toolsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-tools/nuget/v2"
+Register-PSRepository -Name azure-sdk-tools-feed -SourceLocation $toolsFeed -PublishLocation $toolsFeed -InstallationPolicy Trusted -ErrorAction SilentlyContinue
+Install-Module -Repository azure-sdk-tools-feed powershell-yaml
+
+$allowListFilePath = Join-Path $AllowListLocation "${ServiceDirectory}.yml"
```
Comment (benbp) [inline eng/common/scripts/FilterPoliCheckResults.ps1:34]: For my understanding: where is this file coming from? Do we need to worry about cross-language naming overlaps and/or service directory [paths like](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/eventhub/event-hubs) `eventhub/event-hubs` causing issues?

Comment (chidozieononiwu) [reply eng/common/scripts/FilterPoliCheckResults.ps1:34]: That won't be a problem there is a directory for each language.

File: eng/common/scripts/FilterPoliCheckResults.ps1 (hunk 2 above)
Comment (chidozieononiwu) [reply eng/common/scripts/FilterPoliCheckResults.ps1:43]: Is this gonna work. Won't $allowListFilePath just pring as a literal.

Comment (benbp) [reply eng/common/scripts/FilterPoliCheckResults.ps1:43]:
```
$ $foo = "foostring"
$ Write-Host "the value is $foo"
the value is foostring
```

Curly braces are only needed in cases where there isn't a clear word boundary between the variable and subsequent characters, or where there are special characters in the variable name.

https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-string-substitutions?view=powershell-7.1#delineation-with-braces

File: eng/common/scripts/FilterPoliCheckResults.ps1 (hunk 4 above)
Comment (benbp) [reply eng/common/scripts/FilterPoliCheckResults.ps1:34]: What about ServiceDirectory values that aren't valid filenames? I guess I was wrong in that there are no examples of these in the tier 1 language repos, though there are many in the go repo. I think we could either split it in `/`, or just wait and see if we ever add policheck to go and deal with it then.

File: eng/common/scripts/FilterPoliCheckResults.ps1 [hunk 5]
```diff
@@ -3,38 +3,45 @@
...
 
 . "${PSScriptRoot}\logging.ps1"
 
-$RepoRoot = Resolve-Path -Path "${PSScriptRoot}\..\..\..\"
-$PathToAllowListFiles = Join-Path $RepoRoot $ServiceDirtectory
-$PolicCheckAllowListFiles = Get-ChildItem -Path $PathToAllowListFiles -Recurse -File -Include "PoliCheckAllowList.yml"
+# Install Powershell Yaml
+$ProgressPreference = "SilentlyContinue"
+$toolsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-tools/nuget/v2"
+Register-PSRepository -Name azure-sdk-tools-feed -SourceLocation $toolsFeed -PublishLocation $toolsFeed -InstallationPolicy Trusted -ErrorAction SilentlyContinue
```
Comment (weshaggard) [inline eng/common/scripts/FilterPoliCheckResults.ps1:31]: Lets try to use the helper functions we have in https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Helpers/PSModule-Helpers.ps1

File: eng/common/scripts/FilterPoliCheckResults.ps1 [hunk 6]
```diff
@@ -3,38 +3,45 @@
...
   [Parameter(Mandatory=$true)]
   [String] $PoliCheckResultFilePath,
-  [String] $ServiceDirtectory
+  [Parameter(Mandatory=$true)]
+  [String] $ServiceDirectory,
+  [Parameter(Mandatory=$true)]
+  [String] $AllowListLocation
 )
 
 . "${PSScriptRoot}\logging.ps1"
```
Comment (weshaggard) [inline eng/common/scripts/FilterPoliCheckResults.ps1:26]: Any reason we should not import common.ps1?

Comment (chidozieononiwu) [reply eng/common/scripts/FilterPoliCheckResults.ps1:26]: Nope, its just that only the logging is being used at the moment.

File: eng/common/scripts/FilterPoliCheckResults.ps1 (hunk 4 above)
Comment (chidozieononiwu) [reply eng/common/scripts/FilterPoliCheckResults.ps1:34]: Yes, will probably spit it when the time comes to add go.

**Comment by chidozieononiwu** [comment]
Closing this for now since that was not used.

**Pull request closed without merging**